spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/logging/StackdriverLoggingAutoConfigurationTests.java

```diff
@@ -32,107 +34,100 @@
 import org.springframework.test.context.junit4.SpringRunner;
 
 import static org.hamcrest.Matchers.is;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.assertTrue;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /**
  * @author Mike Eltsufin
+ * @author João André Martins
  */
-@RunWith(SpringRunner.class)
-@SpringBootTest(classes = { StackdriverLoggingAutoConfiguration.class },
-		properties = {
-		"debug",
-		"spring.cloud.gcp.config.enabled=false"
-		},
-		webEnvironment = SpringBootTest.WebEnvironment.MOCK)
-public abstract class StackdriverLoggingAutoConfigurationTests {
+public class StackdriverLoggingAutoConfigurationTests {
 
-	@Autowired
-	private Object[] interceptors;
+	private ApplicationContextRunner contextRunner =
+			new ApplicationContextRunner()
+					.withConfiguration(
+							AutoConfigurations.of(StackdriverLoggingAutoConfiguration.class));
 
-	public abstract void test();
-
-	public int countTraceIdInterceptors() {
+	private int countTraceIdInterceptors(Object[] interceptors) {
 		int count = 0;
-		for (int i = 0; i < this.interceptors.length; i++) {
-			if (this.interceptors[i] instanceof TraceIdLoggingWebMvcInterceptor) {
+		for (int i = 0; i < interceptors.length; i++) {
+			if (interceptors[i] instanceof TraceIdLoggingWebMvcInterceptor) {
 				count++;
 			}
 		}
 		return count;
 	}
 
+	@Test
+	public void testDisabledConfiguration() {
+		this.contextRunner.withPropertyValues("spring.cloud.gcp.logging.enabled=false")
+				.run(context -> {
+					Object[] interceptors = context.getBean(Object[].class);
```

Comment:
@artembilan there are no interceptors (i.e., `Object[]` object) in the context. I guess this is because we don't have a Spring Boot test with `webEnvironment = SpringBootTest.WebEnvironment.MOCK` to create the web server and add interceptors to it?

In that case, I guess this is an integration test and we can't make it use `ApplicationContextRunner`?

Reply:
I'm not sure what `interceptors` you are talking about, but there is `MockMVC` if you don't want to use Spring Boot: https://docs.spring.io/spring/docs/5.0.4.RELEASE/spring-framework-reference/testing.html#spring-mvc-test-framework

However I think that `SpringBootTest.WebEnvironment.MOCK` is fully based on the mentioned testing framework. 

I guess your point is do not blow up the whole Spring Boot auto-config for this test, so yes, consider to use `MockMvc` to emulate a web environment for this test.

Reply:
See my latest commit. Unfortunately, couldn't get it to work.

I'm a little out of my depth here, so I might just revert the test to its old format (and maybe rename it to something like `StackdriverLoggingIntegrationTests`).

Reply:
What do you mean with the "See my latest commit"? 
I don't see any explanation of the problem in the commit message.
Sorry

Reply:
Why do you try to get this weird `Object[]` bean? 
Looks like you would like to assert for the `TraceIdLoggingWebMvcInterceptor`.
Why don't call then `context.getBeansOfType(TraceIdLoggingWebMvcInterceptor.class)` ?

Reply:
Sorry, just realised I asked you to look at the latest commit without having pushed it. :(
In any case, thanks for your suggestion. It makes total sense.